framework_tvb/tvb/tests/framework/core/services/project_structure_test.py

```diff
@@ -339,92 +342,56 @@ def test_get_inputs_for_operation(self):
         assert len(inputs) == 1, "Incorrect number of inputs."
         assert conn.id == inputs[0].id, "Retrieved wrong input dataType."
 
-    def test_get_inputs_for_op_group(self, datatype_group_factory, test_adapter_factory):
+    def test_get_inputs_for_group(self, datatype_group_factory, test_adapter_factory):
         """
         Tests method get_datatypes_inputs_for_operation_group.
-        The DataType inputs will be from a DataType group.
         """
-        group = datatype_group_factory(project=self.test_project)
-        datatypes = dao.get_datatypes_from_datatype_group(group.id)
+        zip_path = os.path.join(os.path.dirname(tvb_data.__file__), 'connectivity', 'connectivity_66.zip')
+        conn = TestFactory.import_zip_connectivity(self.test_user, self.test_project, zip_path)
+
+        group = OperationGroup(self.test_project.id, "group", "range1[1..2]")
+        group = dao.store_entity(group)
+
+        view_model = BaseBCTModel()
+        view_model.connectivity = conn.gid
+        adapter1 = ABCAdapter.build_adapter_from_class(TransitivityBinaryDirected)
+
+        algorithm1 = adapter1.stored_adapter
+        algorithm2 = adapter1.stored_adapter
+
+        conn.visible = False
+        dao.store_entity(conn)
+
+        operation1 = Operation(self.test_user.id, self.test_project.id, algorithm1.id,
+                              json.dumps({'gid': view_model.gid.hex}), op_group_id=group.id)
+        operation1.fk_operation_grup = group.id
 
-        datatypes[0].visible = False
-        dao.store_entity(datatypes[0])
-        datatypes[1].visible = False
-        dao.store_entity(datatypes[1])
+        operation2 = Operation(self.test_user.id, self.test_project.id, algorithm2.id,
+                              json.dumps({'gid': view_model.gid.hex}), op_group_id=group.id)
+        operation2.fk_operation_grup = group.id
 
-        op_group = OperationGroup(self.test_project.id, "group", "range1[1..2]")
-        op_group = dao.store_entity(op_group)
-        params_1 = json.dumps({"param_5": "1", "param_1": datatypes[0].gid, "param_6": "2"})
-        params_2 = json.dumps({"param_5": "1", "param_4": datatypes[1].gid, "param_6": "5"})
+        dao.store_entities([operation1, operation2])
 
-        algo = test_adapter_factory(adapter_class=TestAdapter3)
-        op1 = Operation(self.test_user.id, self.test_project.id, algo.id, params_1, op_group_id=op_group.id)
-        op2 = Operation(self.test_user.id, self.test_project.id, algo.id, params_2, op_group_id=op_group.id)
-        dao.store_entities([op1, op2])
+        OperationService()._store_view_model(operation1, dao.get_project_by_id(self.test_project.id), view_model)
+        OperationService()._store_view_model(operation2, dao.get_project_by_id(self.test_project.id), view_model)
```

Comment:
The test might pass now, but the preparation of data for it is hard-coded in the unit-test.
Could you maybe use directly  `SimulatorService.async_launch_and_prepare_pse` instead ?